src/js/components/DSLInputField.js

```diff
@@ -72,7 +96,18 @@ class DSLInputField extends React.Component {
    * @param {SyntheticEvent} event - The change event
    */
   handleChange({target}) {
-    this.props.onChange(new DSLExpression(target.value));
+    this.setState({
+      expression: new DSLExpression(target.value)
```

Comment:
Why are we storing this in state as well? Couldn't we create call a callback to pass it to the parent component that holds the expression?
This way we don't have two sources of truth and are sure for it to not get out of sync